react/components/ManageGroup/index.js

```diff
@@ -43,15 +43,19 @@ class ManageGroup extends Component {
     }).isRequired,
   }
 
+  static getDerivedStateFromProps(nextProps) {
+    const { data: { loading } } = nextProps;
+    if (loading) return null;
+
+    const { data: { group: { name } } } = nextProps;
+    return { name };
+  }
+
```

Comment:
This is one of those things that's going to come up a lot; there's a new static lifecycle method in React called `getDerivedStateFromProps` that is VERY useful here for figuring out state from the cached data.